database/sql/src/main/scala/cromwell/database/slick/WorkflowStoreSlickDatabase.scala

```diff
@@ -13,19 +15,29 @@ trait WorkflowStoreSlickDatabase extends WorkflowStoreSqlDatabase {
 
   import dataAccess.driver.api._
 
-  override def updateWorkflowsInState(updates: List[(String, String)])
-                                     (implicit ec: ExecutionContext): Future[Unit] = {
-    val updateQueries = updates.map({
-      case (oldState, newState) => dataAccess.workflowStateForWorkflowState(oldState).update(newState)
-    })
+  override def setAllRunningToAborting()
+                                      (implicit ec: ExecutionContext): Future[Unit] = {
+    val action = dataAccess
+      .workflowStateForWorkflowState(WorkflowStoreState.Running)
+      .update(WorkflowStoreState.Aborting)
+
+    runTransaction(action) void
+  }
+
+  override def markRunningAndAbortingAsRestarted()
+                                                (implicit ec: ExecutionContext): Future[Unit] = {
+    val action = dataAccess.restartedFlagForRunningAndAborting.update(true)
 
-    val action = DBIO.sequence(updateQueries)
     runTransaction(action) void
   }
 
-  override def updateWorkflowState(workflowId: String, newWorkflowState: String)
-                                     (implicit ec: ExecutionContext): Future[Int] = {
-    val action = dataAccess.workflowStateForId(workflowId).update(newWorkflowState)
+  override def setToAborting(workflowId: String)
+                            (implicit ec: ExecutionContext): Future[Option[Boolean]] = {
+    val action =  for {
+      restarted <- dataAccess.workflowRestartedForId(workflowId).result.headOption
+      _ <- dataAccess.workflowStateForId(workflowId).update(WorkflowStoreState.Aborting)
+    } yield restarted
```

Comment:
Could you comment what `setToAborting` has to do with `"restarting"`?

Reply:
There's a comment in the method definition in the trait but I'll copy it here : 
```
/**
    * Set the workflow to aborting state.
    * @return Some(restarted) if the workflow exists in the store, where restarted is the value of its restarted flag
    *         None if the workflow does not exist in the store
    */
```
We need to know if this workflow is being restarted because if it's not and we haven't started it yet then we can just consider it aborted, but if is, we need to leave it there and pick it up to reconnect to running jobs (while making sure we know we want to abort it)